### DISCUSSION TITLE
Way to detect age key type of an encrypted file?

### DISCUSSION THREAD

**Discussion by e-zk** [FiloSottile/age#220] [Q&A]
To determine the key type of an encrypted secret, currently I am reading the file's header.  
If you open an encrypted file in a text editor it will clearly show `-> ssh-rsa ...`, `-> scrypt ...`, etc on the second line.

Is this the proper way of determining a secret's key type (reading the raw header of the secret, and checking it) or is there a function within the spec/library to do this for me?

Or is it instead advised that I should enforce a specific key type for secrets in my projects?

**Comment by codesoap** [comment]
This issue overlaps with #56.

**Comment by FiloSottile** [comment]
API and CLI ways to inspect an encrypted file are indeed something we'll need to add at some point after v1.0.0. Tracking in #56.

> Or is it instead advised that I should enforce a specific key type for secrets in my projects?

In general, I do recommend that most projects pick a single recipient type, usually the native X25519 keys.
